internal/storage/memory.go

```diff
@@ -267,13 +269,44 @@ func (m *MemoryConnector) DeleteLogs(logs []common.Log) error {
 }
 
 func (m *MemoryConnector) InsertBlockData(data []common.BlockData) error {
+	for _, blockData := range data {
+		dataJson, err := json.Marshal(blockData)
+		if err != nil {
+			return err
+		}
+		m.cache.Add(fmt.Sprintf("blockData:%s", blockData.Block.Number.String()), string(dataJson))
+	}
 	return nil
 }
 
-func (m *MemoryConnector) GetBlockData(blockNumbers []*big.Int) (data []common.BlockData, err error) {
-	return nil, nil
+func (m *MemoryConnector) GetBlockData(qf QueryFilter) ([]common.BlockData, error) {
+	blockData := []common.BlockData{}
+	limit := getLimit(qf)
+	blockNumbersToCheck := getBlockNumbersToCheck(qf)
+
+	for _, key := range m.cache.Keys() {
```

Comment:
Not sure how fast are the requests to the memory cache. Hence this might need parallel execution

Reply:
Memory is as fast as it gets for storage. I don't think there is much to gain from parallelization